### ISSUE TITLE
R session crash when using dbWriteTable

### ISSUE THREAD

**Issue by AdrienMtgn** [r-dbi/RMariaDB#79]
I'm writting tables from R to a database using lattest CRAN version and dev version of RMariaDB for some reasons, most of the time my R session crash.
```
> sessionInfo()
R version 3.4.1 (2017-06-30)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows >= 8 x64 (build 9200)

Matrix products: default

locale:
[1] LC_COLLATE=French_France.1252  LC_CTYPE=French_France.1252    LC_MONETARY=French_France.1252 LC_NUMERIC=C                  
[5] LC_TIME=French_France.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] RMariaDB_1.0-4

loaded via a namespace (and not attached):
[1] bit_1.1-12      compiler_3.4.1  hms_0.4.0       DBI_0.7-13      tools_3.4.1     Rcpp_0.12.14    bit64_0.9-7     pkgconfig_2.0.1 rlang_0.1.4    
```

I used deparse to create the two folowing formulas I use for my reproductible exemple.

[Formula.txt](https://github.com/r-dbi/RMariaDB/files/1576786/Formula.txt)
[Formula2.txt](https://github.com/r-dbi/RMariaDB/files/1576787/Formula2.txt)

Here is the code I use : 

```
library(RMariaDB)
conn<-dbConnect(MariaDB(),user="Ruser",password="pwd",dbname="db",host="192.168.2.12",port=3333)

> dbGetInfo(conn)
$host
[1] "."

$user
[1] "Ruser"

$dbname
[1] "db"

$conType
[1] ". via named pipe"

$serverVersion
[1] "5.6.35-log"

$protocolVersion
[1] 10

$threadId
[1] 3190

$client
[1] "5.5.1"
```

When using the first data.frame :
```
Data<-eval(parse(text = readLines("Formula.txt")))
dbWriteTable(conn,"test",Data,overwrite = T)
```
I get this message : 
![image](https://user-images.githubusercontent.com/29199604/34225346-72b6e088-e5c6-11e7-87cd-a078148e81d8.png)

And when I check the sql database here the written table : 
![image](https://user-images.githubusercontent.com/29199604/34225254-0202ab10-e5c6-11e7-8433-19b44a6161b5.png)

The columns names are written but the table is empty.

for the second one : 
```
Data<-eval(parse(text = readLines("Formula2.txt")))
dbWriteTable(conn,"test",Data,overwrite = T)
```
everything is working and I can find my table in my sql db : 

![image](https://user-images.githubusercontent.com/29199604/34225288-305da96a-e5c6-11e7-83f8-99ca881b6ec9.png)

Log files : 
[rdesktop.log](https://github.com/r-dbi/RMariaDB/files/1576843/rdesktop.log)
[rsession-adrien.montagnon.log](https://github.com/r-dbi/RMariaDB/files/1576844/rsession-adrien.montagnon.log)

Crash Report : (windows dmp file)
[Crash Report.zip](https://github.com/r-dbi/RMariaDB/files/1576849/Crash.Report.zip)

When I use the RMySQL package, everything works smoothly but the Date variables are loaded in the MySQL db as text.

![image](https://user-images.githubusercontent.com/29199604/34226170-2ac031fa-e5c9-11e7-9bd2-adf945c7fe51.png)

![image](https://user-images.githubusercontent.com/29199604/34226281-7edebd88-e5c9-11e7-93db-d043a795041b.png)

**Comment by BramVanRensbergen** [comment]
First, I can reproduce the above error.

I run into (possibly) the same crash each time I attempt to write a dataframe with more than 32 columns. (Even if it only has one row).

The following code always leads to a crash on my system:

> library(RMariaDB)
> con <- dbConnect(RMariaDB::MariaDB(), host = "192.168.2.100", port = 3306, user = "test", dbname = "General")
> dbWriteTable(con, "TEST", as.data.frame(matrix(data = 1, nrow = 1, ncol = 33)), overwrite = T)

Similarly to AdrienMtgn, the table gets created, but no data is entered.

The following DOES work without issue:
> dbWriteTable(con, "TEST", as.data.frame(matrix(data = 1, nrow = 1, ncol = 32)), overwrite = T)

Info: 
sessionInfo()
> R version 3.4.3 (2017-11-30)
> Platform: x86_64-w64-mingw32/x64 (64-bit)
> Running under: Windows 7 x64 (build 7601) Service Pack 1
> 
> Matrix products: default
> 
> locale:
> [1] LC_COLLATE=Dutch_Belgium.1252  LC_CTYPE=Dutch_Belgium.1252    LC_MONETARY=Dutch_Belgium.1252 LC_NUMERIC=C                  
> [5] LC_TIME=Dutch_Belgium.1252    
> 
> attached base packages:
> [1] stats     graphics  grDevices utils     datasets  methods   base     
> 
> other attached packages:
> [1] RMariaDB_1.0-4
> 
> loaded via a namespace (and not attached):
>  [1] bit_1.1-12      compiler_3.4.3  hms_0.4.1       DBI_0.7-14      tools_3.4.3     yaml_2.1.16     Rcpp_0.12.15    bit64_0.9-7     pkgconfig_2.0.1
> [10] rlang_0.1.6  


dbGetInfo(con)

> $host
> [1] "192.168.2.100"
> 
> $user
> [1] "test"
> 
> $dbname
> [1] "General"
> 
> $conType
> [1] "192.168.2.100 via TCP/IP"
> 
> $serverVersion
> [1] "5.5.44-MariaDB"
> 
> $protocolVersion
> [1] 10
> 
> $threadId
> [1] 763
> 
> $client
> [1] "5.5.1"

Any assistance would be greatly appreciated, because writing tables with more than 32 columns is pretty essential for me... Thanks in advance for any help!

**Comment by S-UP** [comment]
I may have the same issue. 

Writing a large (600K rows, 25 variables) data frame to the MySQL server, which is in my case hosted on a remote worksation, always crashes the system.

However, RMySQL performs without issues here.

**Comment by Zelazny7** [comment]
Same issue here. Trying to load 400k rows by 51 columns (1 text, 50 doubles). Creates the table and then promptly crashes Rstudio.

**Comment by krlmlr** [comment]
Thanks. I can't replicate this on Ubuntu, will try Windows. Is this a Windows-only problem?

**Comment by krlmlr** [comment]
Reliably crashes on Windows with the following code:

```R
devtools::load_all()

RMariaDB:::init_logging("VERB")

N <- 100
x <- seq_len(N)
df <- as.data.frame(setNames(as.list(x), paste0("c", x)))

con <- dbConnect(MariaDB(), dbname = "test")
dbWriteTable(con, "test", df, overwrite = TRUE)
dbReadTable(con, "test")
```

**Comment by krlmlr** [comment]
Also confirming the boundary: 32 columns = OK, 33 columns = crash.

**Comment by krlmlr** [comment]
Still seeing the segfault with #82, with libmariadbclient 2.3.4.

**Comment by krlmlr** [comment]
The segfault happens here:

https://github.com/r-dbi/RMariaDB/blob/359b612dd1d613db943489e3f4e8feb99d9f4552/src/MariaResultPrep.cpp#L79

Let's hope libmariadbclient 2.3.5 fixes it. Otherwise we might need to take it upstream.

**Comment by krlmlr** [comment]
Fixed in #92, I can now create and fetch tables with 1000 columns on Windows.

**Comment by Zelazny7** [comment]
Thanks for the fix. Working beautifully now!
